mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdDAO.java

```diff
@@ -357,7 +358,14 @@ private PreparedStatement prepareSelectUidRangeLimited(Session session) {
                         MessageUid.of(row.getLong(IMAP_UID))))
                 .flags(FlagsExtractor.getFlags(row))
                 .modSeq(ModSeq.of(row.getLong(MOD_SEQ_LOWERCASE)))
-                .threadId(ThreadId.fromBaseMessageId(messageIdFactory.of(row.getUUID(THREAD_ID))))
+                .threadId(getThreadIdFromRow(row))
                 .build());
     }
+
+    private ThreadId getThreadIdFromRow(Row row) {
```

Comment:
I would prefer the parameters to be `Row row, MessageId messageId` so that we do not read the messageId twice...